x/pocketcore/keeper/service_test.go

```diff
@@ -107,40 +109,55 @@ func TestKeeper_HandleRelay(t *testing.T) {
 func TestKeeper_HandleRelayWithinTolerance(t *testing.T) {
 	defer gock.Off() // Flush pending mocks after test execution
 
-	// TC - relay proof is within the same session block
-	tc := relayTc{
-		blockHeight:                  976,
-		relayProofSessionBlockHeight: 975,
+	// Each session block is 25 blocks by default genesis params found in DefaultSessionBlocktime, x/nodes/types/params.go
+	// SessionBlockSyncAllowance is 1 session block by default found in DefaultSessionBlockSyncAllowance, types/config.go
+
+	tcs := []relayTc{
+		// tc 1 - relay proof session height matches the same session block (max height boundary)
+		{
+			blockHeight:                  976,
+			relayProofSessionBlockHeight: 976,
+		},
+		// tc2 - relay proof session height is within tolerance (min height boundary)
+		{
+			blockHeight:                  976,
+			relayProofSessionBlockHeight: 951,
+		},
 	}
 
-	gock.New("https://www.google.com:443").
-		Post("/").
-		Reply(200).
-		BodyString("bar")
-	keeper, mockCtx, validRelay := setupHandleRelayTest(t, tc.blockHeight, tc.relayProofSessionBlockHeight)
-	resp, err := keeper.HandleRelay(mockCtx, validRelay)
-	assert.Nil(t, err, err)
-	assert.NotNil(t, resp)
-	assert.NotEmpty(t, resp)
-	assert.Equal(t, resp.Response, "bar")
+	for _, tc := range tcs {
+		gock.New("https://www.google.com:443").
+			Post("/").
+			Reply(200).
+			BodyString("bar")
+		keeper, mockCtx, validRelay := setupHandleRelayTest(t, tc.blockHeight, tc.relayProofSessionBlockHeight)
+		resp, err := keeper.HandleRelay(mockCtx, validRelay)
+		assert.Nil(t, err, err)
+		assert.NotNil(t, resp)
+		assert.NotEmpty(t, resp)
+		assert.Equal(t, resp.Response, "bar")
+	}
 }
 
 func TestKeeper_HandleRelayOutsideTolerance(t *testing.T) {
 	// Each session block is 25 blocks by default genesis params found in DefaultSessionBlocktime, pocket-core/x/nodes/types/params.go
 	tcs := []relayTc{
 		{
-			// tc 1: relay proof ahead of current block height
+			// tc 1: relay proof session height is not within tolerance (max boundary exceeded)
 			blockHeight:                  926,
 			relayProofSessionBlockHeight: 976,
 		},
 		{
-			// tc 2: relay proof two sessions behind
-			blockHeight:                  926,
-			relayProofSessionBlockHeight: 900,
+			// tc 2: relay proof session height is not within tolerance (min boundary exceeded)
+			blockHeight:                  976,
+			relayProofSessionBlockHeight: 926,
 		},
 	}
 	clientBlockSyncAllowanceTmp := types.GlobalPocketConfig.ClientBlockSyncAllowance // hold the initial value before mutating, this should not matter anyways since tests concurrency = 1
 	types.GlobalPocketConfig.ClientBlockSyncAllowance = 100                          // Allows the client block sync tolerance to go above 100 blocks for test purposes
```

Comment:
Why do we need it to go above 100 for test purposes even though we don't modify it in `TestKeeper_HandleRelayWithinTolerance`? Ditto for other tests.

Reply:
It was just a conservative number, the default is 10. I can shrink this, but it does not really impact the relay test, unless the ClientBlockSyncAllowance is < 25.